code/HabitTracker/app/src/main/java/com/example/habittracker/activities/eventlist/EventListActivity.java

```diff
@@ -69,41 +91,177 @@ public void onClick(View v) {
+                    System.out.println(habitList.get(i).getTitle());
+                    habit_list.add(habitList.get(i).getTitle());
+                    Log.d("Here", habit_list.get(i));
+                    Log.d("list size", String.valueOf(habitList.size()));
+                }
+                CollectionReference colRef;
+                eventDataList.clear();
+                for (int i = 0;i<habit_list.size();i++) {
+                    colRef = dm.getUsersColRef()
+                            /**********/
+//                            .document("John_test_user")
+                            // Replace with this line when sharedInfo is up
+                            .document(SharedInfo.getInstance().getCurrentUser().getUsername())
+                            /*********/
+                            .collection(habitsColName)
+                            .document(habit_list.get(i))
+                            .collection(habitEventsColName);
+                    colRef.addSnapshotListener(new EventListener<QuerySnapshot>() {
+                        @Override
+                        public void onEvent(@Nullable QuerySnapshot queryDocumentSnapshots, @Nullable
+                                FirebaseFirestoreException error) {
+                            for (QueryDocumentSnapshot doc : queryDocumentSnapshots) {
+                                boolean duplicate_flag = false;
+                                for (int i = 0;i<eventDataList.size();i++) {
+                                    if(eventDataList.get(i).getEventId() == doc.getId()) {
+                                        duplicate_flag = true;
+                                        Log.d("same!", String.valueOf(doc.getId()));
+                                    }
+                                }
+                                if(duplicate_flag) {
+                                    continue;
+                                }
+                                Log.d(DB_TAG, String.valueOf(doc.getData().get("Habit")));
+                                String eventID = doc.getId();
+                                String habitID = (String) doc.getData().get("Habit");
+
+                                Log.d("eventID", String.valueOf(doc.getId()));
+                                Log.d("date", String.valueOf(doc.getData().get("startDate")));
+
+                                ArrayList<Integer> startDate = (ArrayList<Integer>) doc.getData().get("startDate");
+                                String comments = (String) doc.getData().get("comment");
+//                    String location = (String) doc.getData().get("location");
+//                    String image = (String) doc.getData().get("image");
+                                HabitEvent temp = new HabitEvent();
+                                temp.setStartDate(startDate);
+                                temp.setComment(comments);
+                                temp.setEventId(eventID);
```

Comment:
One thing to note. What happens to the UI when non-mandatory fields are null?
ie. what if there is no comment. 
Did you test that sort of thing?

Reply:
Yes, I have test those special cases. Nothing would happen, comment field is empty.

Reply:
Awesome!! Thanks for testing that out.